sectionproperties/pre/sections.py

```diff
@@ -1499,11 +1491,16 @@ def offset_perimeter(self, amount: float = 0, where = "exterior", resolution: fl
             compound = geometry_2.align_center(geometry_1).align_to(geometry_1, on="top") + geometry_1
             new_geometry = compound.offset_section_perimeter(amount=-3)
         """
-        geoms_acc = []
-        for geom in self.geoms:
-            geoms_acc.append(geom.offset_perimeter(amount, where, resolution))
-        new_geom = CompoundGeometry(geoms_acc)
-        return new_geom
+        # geoms_acc = []
+        # for geom in self.geoms:
+        #     geoms_acc.append(geom.offset_perimeter(amount, where, resolution))
+        # new_geom = CompoundGeometry(geoms_acc)
+        # return new_geom
+
+        unionized_poly = unary_union([geom.geom for geom in self.geoms])
```

Comment:
Taking a leaf out of the `CompoundGeometry.calculate_perimeter()` book, seems to convert to a single `Polygon` object which we can then use to offset as per the `Geometry` method?

Reply:
I have an idea...